dsa/tests/signature.rs

```diff
@@ -22,19 +23,19 @@ const MESSAGE: &[u8] = b"test";
 ///
 /// This signature was generated using the keys generated by this CSPRNG (the per-message `k` component was also generated using the CSPRNG)
 const MESSAGE_SIGNATURE_CRATE_ASN1: &[u8] = &[
-    0x30, 0x2C, 0x02, 0x14, 0x45, 0x1D, 0xE5, 0x76, 0x21, 0xD8, 0xFD, 0x76, 0xC1, 0x6F, 0x45, 0x4E,
-    0xDE, 0x5F, 0x09, 0x79, 0x76, 0x52, 0xF3, 0xA5, 0x02, 0x14, 0x53, 0x60, 0xE6, 0xB7, 0xF0, 0xCF,
-    0xAE, 0x49, 0xB1, 0x58, 0x5C, 0xCF, 0x5F, 0x3F, 0x94, 0x49, 0x21, 0xA0, 0xBF, 0xD2,
+    0x30, 0x2d, 0x2, 0x14, 0x4, 0x23, 0xc3, 0xf1, 0xa5, 0x15, 0x65, 0xe5, 0xcc, 0x30, 0xdd, 0x55,
+    0x96, 0x27, 0x30, 0xfb, 0x42, 0xf9, 0x2a, 0x6b, 0x2, 0x15, 0x0, 0x95, 0xd9, 0x42, 0xc2, 0xab,
+    0xe7, 0x62, 0x49, 0x80, 0x4f, 0x53, 0xf1, 0x37, 0x57, 0x81, 0x7b, 0xea, 0x71, 0xc9, 0x2f,
 ];
 
 /// Message signed by OpenSSL using the keys generated by this CSPRNG
 ///
 /// This signature was generated using the SHA-256 digest
-const MESSAGE_SIGNATURE_OPENSSL_ASN1: &[u8] = &[
-    0x30, 0x2C, 0x02, 0x14, 0x6D, 0xB3, 0x8E, 0xAF, 0x97, 0x13, 0x7E, 0x07, 0xFF, 0x24, 0xB8, 0x66,
-    0x97, 0x18, 0xE1, 0x6F, 0xD7, 0x9A, 0x28, 0x2D, 0x02, 0x14, 0x47, 0x8C, 0x0B, 0x96, 0x51, 0x08,
-    0x08, 0xC8, 0x34, 0x9D, 0x0D, 0x41, 0xC7, 0x73, 0x0F, 0xB5, 0x9C, 0xBB, 0x00, 0x34,
-];
+const MESSAGE_SIGNATURE_OPENSSL_ASN1: &[u8] = &hex!(
+    "302e 0215 0092 4523 9462 f59b 5936 d355
+     8b16 8dbb 0915 7ba1 5302 1500 9d8d 7aa7
+     02a6 2dde 1975 e9f2 a20c 73b4 7dd3 0912"
+);
```

Comment:
The behavior of `Components::generate` changed because of the switch to crypto-bigint (and crypto-primes). This makes the "deterministic" private key used for test change. And as a result, makes the signatures change.

I don't believe we offered any guarantee towards the stability of the `Component::generate`?

Reply:
No, we didn't make any guarantees. The only guarantee we provided was "it outputs something that works".  

If some code relies on this staying the same it's really fragile anyway, since they'd have to use the same algorithms to generate primes and factors, which can differ between implementations. So yeah, this should be fine.